sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsMeta.scala

```diff
@@ -712,7 +712,8 @@ object ExpressionContext {
     val parent = findParentPlanMeta(meta)
     assert(parent.isDefined, "It is expected that an aggregate function is a child of a SparkPlan")
     parent.get.wrapped match {
-      case _: WindowExecBase => WindowAggExprContext
+      case agg if ShimLoader.getSparkShims.isWindowFunctionExec(agg.asInstanceOf[SparkPlan]) =>
```

Comment:
nit: this would not match on `agg` if it somehow is not a `SparkPlan`, and remove the explicit `asInstanceOf`

```
case agg: SparkPlan if ShimLoader.getSparkShims.isWindowFunctionExec(agg) =>
```

Reply:
> if it somehow is not a `SparkPlan`.

I'm attempting to catch an `Exec`, i.e. either `WindowExec` or `RunningWindowFunctionExec`. Their most common base class seems to be `SparkPlan`. I wouldn't be able to handle a non-`SparkPlan` anyway. :]

> remove the explicit `asInstanceOf`

Thanks. Let me try that.

Reply:
> if it somehow is not a `SparkPlan`.

I'm attempting to catch an `Exec`, i.e. either `WindowExec` or `RunningWindowFunctionExec`. Their most common base class seems to be `SparkPlan`. I wouldn't be able to handle a non-`SparkPlan` anyway. :]

> remove the explicit `asInstanceOf`

Thanks. Let me try that.

Reply:
> Thanks. Let me try that.

Actually, no, I can't. `parent.get.wrapped` is apparently of type `Any`. And `isWindowFunctionExec()` expects a `SparkPlan`.  :/

Reply:
@abellina: I can't seem to avoid the cast.Perhaps I've misunderstood the suggestion?

Reply:
When you match `agg: SparkPlan`, `agg` essentially becomes: `parent.get.wrapped.asInstanceOf[SparkPlan]`. Did you get a compilation error?

Reply:
Ok, I stand corrected. I tried a simpler example than this. I think `agg` is `SparkPlan` only in the body of the case. Never mind :)

Reply:
Yep, I misunderstood the suggestion. Let me try this. Will update shortly.

Reply:
Yep, this is cleaner. Thanks, @abellina. Sorry I misread it the first time.